### ISSUE TITLE
AI Failure to respond

### ISSUE THREAD

**Issue by MetricCepheid** [shasankp000/AI-Player#9]
I'm having an issue where I have ollama and llama2 downloaded but I'm getting nowhere with the AI. It gives me the entire conversation spiel but then I try to talk to it and it just gives me an error. A screenshot provided of the issue, alongside logs from both Minecraft and Ollama plus a video.

https://youtu.be/V_1ePbahdIE

![image](https://github.com/user-attachments/assets/e1ea82e9-bc81-43f4-8761-cd2fb3fe0cbf)
[latest.log](https://github.com/user-attachments/files/17640937/latest.log)
[app.log](https://github.com/user-attachments/files/17640939/app.log)
[server.log](https://github.com/user-attachments/files/17640940/server.log)

**Comment by MetricCepheid** [comment]
Disregard. Latest hotfix I've found on the releases page talks about this processing message problem.

**Comment by MetricCepheid** [comment]
Undisregard, even after installing Gemma2, I'm still having this issue. Supplied with new sets of logs.

[latest.log](https://github.com/user-attachments/files/17641069/latest.log)
[app.log](https://github.com/user-attachments/files/17641064/app.log)
[server.log](https://github.com/user-attachments/files/17641065/server.log)

![image](https://github.com/user-attachments/assets/cd4bbea6-53b0-48bf-b10d-a0879e3a7fde)

**Comment by shasankp000** [comment]
You need to configure the language model in `/configMan`. Set it to gemma2

**Comment by MetricCepheid** [comment]
Here is a full video showcasing the steps I go through with switching the bot to Gemma2 and still coming down to this problem.

https://youtu.be/ONo30Ceh258

**Comment by MetricCepheid** [comment]
Tried it again on a new world thinking that could be an issue but same issue pretty much.

[latest.log](https://github.com/user-attachments/files/17654397/latest.log)
[app.log](https://github.com/user-attachments/files/17654408/app.log)
[server.log](https://github.com/user-attachments/files/17654409/server.log)

**Comment by MetricCepheid** [comment]
It's not lagging super bad, it hitches but genuinely I haven't had any problem and yes, I am running ollama locally in the background- I am also running the most recent version of the mod from the releases page on this github.

**Comment by shasankp000** [comment]
Yeah then I think it's your pc's issue because of the "hitches" you mentioned I guess the mod fails to reach the ollama server.
Thing is I made the mod too dependent on every small conversation related task on the language model so could be the fact that your pc is hitting a memory cap.

If that's the case then please wait till next month, I have my exams going on right now, and I will be able to resume active development after that. I already have a lot of fixes and updates planned for the mod.

**Comment by shasankp000** [comment]
You can try allocating a bit more ram to minecraft and then try it out

**Comment by NeonFast** [comment]
> Here is a full video showcasing the steps I go through with switching the bot to Gemma2 and still coming down to this problem.
> 
> https://youtu.be/ONo30Ceh258

tryed by nothing

**Comment by shasankp000** [comment]
@NeonFast please wait till mid of December for my exams to end after that I can resume active development and roll out updates and bug-fixes, especially for this one.

Thank you and @MetricCepheid for taking an interest in the mod.

**Comment by shasankp000** [comment]
Keeping issue opened till this bug is resolved for everyone.

**Comment by shasankp000** [comment]
@NeonFast @MetricCepheid This issue should be fixed in the latest release. Check out : https://modrinth.com/mod/ai-player/version/1.0.3-alpha-1

**Comment by MetricCepheid** [comment]
Is it known that the bot actively doesn't follow directions and just outputs what it thinks it's doing? Also how did your exams go?

**Comment by shasankp000** [comment]
@MetricCepheid  well  I haven't implemented a "goal-oriented" learning yet. I just implemented reinforcement learning which makes the bot able to detect, react and learn about certain situations in it's environment. Right now it only does so for hostile mobs around it.

The detection triggers are hardcoded, since minecraft is finite so the situations will a finite number as well.

For the bot to "follow-directions" I need to implement a goal-oriented learning mechanism where the bot will do random, stuff and see if it's goal is achieved or not and overtime using the learning algorithm I made, train itself to follow that specific goal.

However the bot can follow some tasks, since it's hardcoded, like detect blocks, go to an XZ coords, but only the methods to do that are hardcoded, you can ask the bot naturally as you would to do that task.

In the coming patches I will fully integrate the learning mechanism with the environment triggers I made along with the language model to make a fully synchronised AI.


One more thing about reinforcement learning is that this is unsupervised learning. The bot will do initially random stuff and over time learn what to do in certain scenarios.



Also my exams got postponed to January, thanks for asking.

**Comment by shasankp000** [comment]
@MetricCepheid Have you experienced the issue anymore? If not then do let me know, so that I can close this issue.

**Comment by MetricCepheid** [comment]
I have not, thank you.

**Comment by foss-lover34** [comment]
@shasankp000 
hey man i really love this mod and wish to get it working... i know you're busy and all, but i am getting this constant issue of "i am unable to understand your message" just like in his screenshot.
ollama server running in background, its executables not blocked by my firewall. Installed from OllamaInstaller.exe from their website. 
Ollama cmd binaries downloaded, run succeffully all the commands in the tutorial, i can run the model with "ollama.exe run llama3.2".
installed the dependencies and spawn the bot succefully, configured the llm with /configman. i just cant get it to work. please help me out

**Comment by shasankp000** [comment]
Yeah unfortunately this is not the issue of the mod, it's the LLM genuinely unable to understand. I have done a thorough research within my code and this happens sometimes when llama3.2 is just unable to map the NLP (Natural language processing) instructions correctly.

Right now there is no fix to this but in the next upcoming hotfix of version 1.0.3-alpha-2 I will introduce deepseek-r1's 3B model, which should perform far better.

**Comment by foss-lover34** [comment]
wow. that sounds freaking amazing bro. so you confirm the i cant talk to the AI plyer currently (i tried with a simple hey or "answer with yes"),  but also why can i run the model in cmd and those prompts work fine but the bot in MC cant?

**Comment by foss-lover34** [comment]
also do you think you will update the mod to 1.21? dont need that just curious. also thanks for answering bro @shasankp000

**Comment by foss-lover34** [comment]
thats fucking awesome. i just cant wait to try this mod out :)
